src/io/read_tree_lhalo_hdf5.c

```diff
@@ -60,15 +63,43 @@ int setup_forests_io_lht_hdf5(struct forest_info *forests_info, const int firstf
         }
         int status = fill_metadata_names(&metadata_names, run_params->TreeType);
         if (status != EXIT_SUCCESS) {
-            ABORT(status);
+            return -1;
```

Comment:
:( `return status` please

Reply:
Since this function is returning a status, the caller is checking for "EXIT_SUCCESS" - so I have fixed these to return "EXIT_FAILURE" etc...